sdk/storage/inc/datalake/datalake_utilities.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/crypt.hpp"
+
+#include <map>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  inline std::map<std::string, std::string> ParseDataLakePropertiesString(
+      const std::string& dataLakePropertiesString)
+  {
+    std::map<std::string, std::string> result;
+
+    auto startItr = dataLakePropertiesString.begin();
+    auto endItr = startItr + dataLakePropertiesString.find(",");
+    // a=b,c=d,e=f
+    // 012345678910
+    do
+    {
+      auto equalItr = startItr
+          + dataLakePropertiesString.find(
+              "=", startItr - dataLakePropertiesString.begin(), endItr - startItr);
+      result[std::string(startItr, equalItr - 1)]
+          = Base64Decode(std::string(equalItr + 1, endItr - 1));
+
+      startItr = endItr + 1;
```

Comment:
**Maybe** it's undefined behavior if `endItr` is already end?

Reply:
Will rewrite.